src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -499,11 +521,22 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
                 await singleton.AcquireAsync(functionCancellationTokenSource.Token);
             }
 
-            var timer = StartFunctionTimeout(instance, globalFunctionTimeout, functionCancellationTokenSource, traceWriter);
+            var timer = StartFunctionTimeout(instance, globalFunctionTimeout, timeoutWhileDebugging, functionCancellationTokenSource, traceWriter);
             try
             {
                 await invoker.InvokeAsync(invokeParameters);
             }
+            catch (OperationCanceledException)
+            {
+                // if cancellationToken is not signaled, it means this is a timeout
+                if (throwOnFunctionTimeout && !cancellationToken.IsCancellationRequested)
```

Comment:
This feels a little fragile. Is the issue that you can't throw the exception from the background timer thread? How about using another mechanism for determining whether the timer has expired, e.g. subscribing to the Elapsed with a local delegate that sets a flag you can check?